test/extended/baremetal/high_availability.go

```diff
@@ -0,0 +1,151 @@
+package baremetal
+
+import (
+	"context"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	clusteroperatorhelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	deleteInterval           = 1 * time.Second
+	deleteWaitTimeout        = 3 * time.Minute
+	restartInterval          = 10 * time.Second
+	restartWaitTimeout       = 10 * time.Minute
+	baremetalNamespace       = "openshift-machine-api"
+	clusterBaremetalOperator = "cluster-baremetal-operator"
+	metal3Deployment         = "metal3"
+)
+
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platform should ensure", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("baremetal")
+
+	g.It("cluster baremetal operator and metal3 deployment return back healthy after they are deleted", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("delete cluster baremetal operator")
+		err = c.AppsV1().Deployments(baremetalNamespace).Delete(context.Background(), clusterBaremetalOperator, v1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until cluster baremetal operator is deleted")
+		err = wait.PollImmediate(deleteInterval, deleteWaitTimeout, func() (bool, error) {
+			_, err := c.AppsV1().Deployments(baremetalNamespace).Get(context.Background(), clusterBaremetalOperator, v1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					return true, nil
+				}
+
+				e2e.Logf("Error getting cluster baremetal operator deployment: %v", err)
+				return false, nil
+			}
+
+			return false, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until cluster baremetal operator returns back healthy")
+		err = wait.Poll(restartInterval, restartWaitTimeout, func() (bool, error) {
+			dc, err := c.AppsV1().Deployments(baremetalNamespace).Get(context.Background(), clusterBaremetalOperator, v1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					return false, nil
+				}
+
+				e2e.Logf("Error getting cluster baremetal operator deployment: %v", err)
+				return false, nil
+			}
+
+			if dc.Status.AvailableReplicas != 1 {
+				return false, nil
+			}
+
+			baremetalCO, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(context.Background(), "baremetal", v1.GetOptions{})
+			if err != nil {
+				e2e.Logf("Error getting authentication operator: %v", err)
```

Comment:
authentication operator?  or baremetal operator?

Reply:
It should definitely be baremetal operator.